File: 563backpackV/backpackV.py
import itertools

class Solution:
    """
    @param nums: an integer array and all positive numbers
    @param target: An integer
    @return: An integer
    """
    def backPackV(self, nums, target):
        # write your code here
        dp = [1, *itertools.repeat(0, target)]
        for item in nums:
            for size in range(target, item - 1, -1): dp[size] += dp[size - item]
        return dp[-1]

File: 563backpackV/backpackV.cpp
class Solution {
public:
    /**
     * @param nums: an integer array and all positive numbers
     * @param target: An integer
     * @return: An integer
     */
    int backPackV(vector<int> &nums, int target) {
        // write your code here
        std:;vector<int> dp(target + 1);
        dp.front() = 1;
        for (auto const item: nums)
            for (std::size_t size{target}; size > item - 1; --size) dp[size] += dp[size - item];
        return dp.back();
    }
};
